app/services/integration/kubernetes_service.rb

```diff
@@ -135,18 +135,19 @@ def initialize(attributes, **options)
   class RouteSpec < K8s::Resource
     def initialize(url, service, port)
       uri = URI(url)
+      tls_options = {
+        insecureEdgeTerminationPolicy: 'Redirect',
+        termination: 'edge'
+      } if uri.class == URI::HTTPS
+
       super({
         host: uri.host || uri.path,
         port: { targetPort: port },
-        tls: {
-          insecureEdgeTerminationPolicy: 'Redirect',
-          termination: 'edge'
-        },
         to: {
           kind: 'Service',
           name: service
         }
-      })
+      }.merge(tls: tls_options))
```

Comment:
Is it OK to merge `tls: nil` if `uri.class != URI::HTTPS`?

Reply:
Yes that is what k8s expect, there is a test for it

Anyway this PR is in conflict with the `KUBERNETES_ROUTE_TLS` env
I will think of a way to make it compatible later